src/DockerRunner.jl

```diff
@@ -93,9 +93,18 @@ function DockerRunner(workspace_root::String;
     push!(workspaces, compiler_wrapper_path => "/opt/bin")
 
     # Generate CMake and Meson files
-    generate_toolchain_files!(platform, envs; toolchains_path=toolchains_path)
+    generate_toolchain_files!(platform, envs, toolchains_path)
     push!(workspaces, toolchains_path => "/opt/toolchains")
 
+    # Generate directory where to write Cargo config files
+    if :rust in only(extract_kwargs(kwargs, (:compilers,))).second
+        cargo_dir = mktempdir()
+        cargo_config_file!(cargo_dir)
+        # Add to the list of mappings a subdirectory of ${CARGO_HOME}, whose content
+        # will be put in ${CARGO_HOME}.
+        push!(mappings, cargo_dir => envs["CARGO_HOME"] * "/" * randstring())
```

Comment:
Do we want to preserve the `mappings` argument and create a copy of it, instead of overwriting it?  I don't think this argument is super critical though.